### DISCUSSION TITLE
How to make the edit and drag mode work only for group draw items?

### DISCUSSION THREAD

**Discussion by Mackiavelly** [geoman-io/leaflet-geoman#901] [Questions & Help] [answered]
```
let drawnItems = new L.FeatureGroup(),
	pmOptions = {
		layerGroup     : drawnItems,
	};
map.pm.setGlobalOptions(pmOptions);
```
How to make the edit and drag mode work only for this group (drawnItems)?

**Comment by Falke-Design** [comment]
## If drawnItems contains only layers that created by Geoman
One way is to set on all not drawnItems the option `pmIgnore: true` => `layer.setStyle({pmIgnore: true});`. 

For GeoJSON Layers:
```
var geo = L.geoJSON(json, {
    pmIgnore: true
  }).addTo(map); // Other not editable layers
  map.fitBounds(geo.getBounds());

  let drawnItems = new L.FeatureGroup().addTo(map),
    pmOptions = {
      layerGroup: drawnItems,
    };
  map.pm.setGlobalOptions(pmOptions);
```

-----

Or you set `L.PM.setOptIn(true)` and then no layer will work with Geoman until you set on a layer `pmIgnore: false` and re-init it with `L.PM.reInitLayer(layer);`

For GeoJSON:
```
  L.PM.setOptIn(true); // Disable Geoman for all layers by default
  var geo = L.geoJSON(json).addTo(map);  // Other not editable layers
  map.fitBounds(geo.getBounds());

  let drawnItems = new L.FeatureGroup().addTo(map),
    pmOptions = {
      layerGroup: drawnItems,
    };
  map.pm.setGlobalOptions(pmOptions);
  drawnItems.options.pmIgnore = false; // Specialy needed for LayerGroups
  L.PM.reInitLayer(drawnItems); // Make the LayerGroup accessable for Geoman

  map.on('pm:create',(e)=>{
    e.layer.setStyle({pmIgnore: false});
    L.PM.reInitLayer(e.layer); // enable Geoman for this special layer
  })
```

## If drawnItems can contains layers not created by Geoman:
You set `L.PM.setOptIn(true)` and then no layer will work with Geoman until you set on a layer `pmIgnore: false` and re-init it with `L.PM.reInitLayer(layer);`

```
  L.PM.setOptIn(true);
  let drawnItems = new L.FeatureGroup().addTo(map),
    pmOptions = {
      layerGroup: drawnItems,
    };
  map.pm.setGlobalOptions(pmOptions);
  drawnItems.options.pmIgnore = false; // Specialy needed for LayerGroups

  // Enable Geoman for all layers in LayerGroup
  drawnItems.setStyle({pmIgnore: false}); // To set pmIgnore on all Layers
  L.PM.reInitLayer(drawnItems); // Make the LayerGroup accessable for Geoman
 
  // Enable all new drawn layers
  map.on('pm:create',(e)=>{
    e.layer.setStyle({pmIgnore: false});
    L.PM.reInitLayer(e.layer);
  })
```

**Reply by Mackiavelly** [reply]
Unfortunately pmIgnore also disables other functions such as snapping and so on, but I only need them to be impossible to move or change the geometry. It needed for situation then I have big map with many layers, and need to draw new item, in process draw I can change geometry and reposition new item, only for drawing item.

**Reply by Falke-Design** [reply]
I understand, currently it is not possible to this, but in the next release there a few new options. We want to release it this month.

<html><body>


Edit option | default | Description
-- | -- | --
allowEditing | true | Edit-Mode for the layer can disabled (pm.enable()).
allowRemoval | true | Removing can be disabled for the layer.
allowCutting | true | Layer can be prevented from cutting.
allowRotation | true | Layer can be prevented from rotation.
draggable | true | Dragging can be disabled for the layer.


</body>
</html>

Then it would look like:
```
map.pm.setGlobalOptions({allowEditing: false}); // To disable editing on all layers
drawnItems.pm.applyOptionsToAllChilds({allowEditing: true}); // Allow editing for all layers in the group
map.pm.enableGlobalEditMode();
```

**Reply by Mackiavelly** [reply]
Its be nice!

**Reply by Mackiavelly** [reply]
> I understand, currently it is not possible to this, but in the next release there a few new options. We want to release it this month.
> 
> Edit option	default	Description
> allowEditing	true	Edit-Mode for the layer can disabled (pm.enable()).
> allowRemoval	true	Removing can be disabled for the layer.
> allowCutting	true	Layer can be prevented from cutting.
> allowRotation	true	Layer can be prevented from rotation.
> draggable	true	Dragging can be disabled for the layer.
> Then it would look like:
> 
> ```
> map.pm.setGlobalOptions({allowEditing: false}); // To disable editing on all layers
> drawnItems.pm.applyOptionsToAllChilds({allowEditing: true}); // Allow editing for all layers in the group
> map.pm.enableGlobalEditMode();
> ```

How it is work if I dynamical add new layers in layerGroups?

**Reply by Falke-Design** [reply]
On the place where you add dynamical a new layer, you have access to the layer object and there you can add options it, and then call `layer.pm.setOptions({allowEditing: true});`

**Comment by Mackiavelly** [comment, marked as answer]
@Falke-Design,
I find this [Falke-Design/PMLock](https://github.com/Falke-Design/PMLock), it's a great, and that I need.

**Reply by Mackiavelly** [reply]
```
// for layer groups use like this
let markers = L.layerGroup();
...
markers.eachLayer(layer => {
	layer.options.pmLock = true;
});
```

**Reply by Falke-Design** [reply]
Is ok but it will be depecrated with the next Release.
